src/transformers/pipelines/base.py

```diff
@@ -902,10 +907,14 @@ def device_placement(self):
             with tf.device("/CPU:0" if self.device == -1 else f"/device:GPU:{self.device}"):
                 yield
         else:
-            if self.device.type == "cuda":
-                torch.cuda.set_device(self.device)
-
-            yield
+            if hasattr(torch.device, "__enter__"):
```

Comment:
nit: we typically check for compatibility with flags like `is_torch_greater_or_equal_than_2_0` in [pytorch_utils.](https://github.com/huggingface/transformers/blob/1815d1865e61df3be1cbe37ee8b78f6017b0b1e2/src/transformers/pytorch_utils.py#L31). It's a bit cleaner than checking for a private attribute and is clearer for the reader what's being checked

Reply:
Ok I can confirm it has been introduced on PT>=2.0.0